### ISSUE TITLE
Suspiciou conditional with Countable instance

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1709]
Hello!

I have found some issues on my code because of misunderstood by my side. Laravel, for instance, have a feature called `Collection`, that is a "improved array", that is an `Countable`. The problem is that a lot of time I check if it contains an element or not as:

```php
if ($items) { ... }
```

But it will always be `true`, because `$items` is, actually an object of kind `Collection`.

**What is my suggestion:**

As this kind of conditional is common (when `$items` could be `null` too, or checking if `array` is empty), in case of the object is an instance of `Countable` (as `Collection` is), so force user double check this conditional by using, for instance, `count()` or something more than just passing the variable name (in the `Collection`, for instance, we could do `$items->isEmpty()`).

**Comment by tlamy** [comment]
This is why i brought up #1603. In the meantime, there once was an inspection that did exactly that, and my Java skills are abysmal.
If anyone wants to take over the PR, feel free to do so.

**Comment by rentalhost** [comment]
@tlamy this is very useful for forcing conditional typing verification, but maybe it’s too much for me. I don't know, maybe I'll try to use it, but it would certainly solve the problem mentioned above!

**Comment by ea-inspections-team** [comment]
Thanks for the PR @tlamy, we are still catching up on the codebase and how it's designed. Once we are comfortable with big changes, we'll get back to it.

**Comment by kalessil** [comment]
@ea-inspections-team, I'd really appreciate it if you could work of this ticket. Thanks!

**Comment by ea-inspections-team** [comment]
@kalessil, @rentalhost : implemented!
